File: Java/0001-两数之和.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
    for (int i = 0, k = 1, n = nums.length; k < n; i++) {
      if (i + k >= n) {
        i = -1;
        k++;
      } else if (target == nums[i] + nums[i + k]) {
        return new int[]{i, i + k};
      }
    }
    return null;
  }
}

File: TypeScript/0001-两数之和.ts
function twoSum(nums: number[], target: number): number[] {
  for (let i: number = 0, k: number = 1, n: number = nums.length; i < n; i++) {
    if (i + k >= n) {
      i = -1;
      k++;
    } else if (nums[i] + nums[i + k] === target) {
      return [i, i + k];
    }
  }
  return [];
}
